src/Data/List/Relation/Unary/All/Properties.agda

```diff
@@ -505,6 +506,52 @@ module _ {P : A → Set p} {Q : A → Set q} (P? : Decidable P) where
   ... | false = filter⁺ Qxs
   ... | true  = Qx ∷ filter⁺ Qxs
 
+  filter⁻ : ∀ {xs} → All Q (filter P? xs) → All Q (filter (¬? ∘ P?) xs) → All Q xs
+  filter⁻ {[]}         []          []                           = []
+  filter⁻ {x ∷ xs}       all⁺        all⁻ with P? x  | ¬? (P? x)
+  filter⁻ {x ∷ xs}       all⁺        all⁻  | yes  Px | yes  ¬Px = contradiction Px ¬Px
+  filter⁻ {x ∷ xs} (qx ∷ all⁺)       all⁻  | yes  Px | no  ¬¬Px = qx ∷ filter⁻ all⁺ all⁻
+  filter⁻ {x ∷ xs}       all⁺  (qx ∷ all⁻) | no    _ | yes  ¬Px = qx ∷ filter⁻ all⁺ all⁻
+  filter⁻ {x ∷ xs}       all⁺        all⁻  | no  ¬Px | no  ¬¬Px = contradiction ¬Px ¬¬Px
+
+------------------------------------------------------------------------
+-- derun and deduplicate
+
+module _ {P : A → Set p} {Q : A → A → Set q} (Q? : B.Decidable Q) where
```

Comment:
`Q` is usually reserved for unary relations. `R` is the usual name for binary relations. This applies generally throughout the PR (just made a note to add it to the style guide).